docs/_includes/Types.md

```diff
@@ -1918,34 +1918,83 @@ Note that when there is no body, Dafny assumes that the *ensures*
 clauses are true without proof.
 
 ### Constructors
-To write structured object-oriented programs, one often relies on that
-objects are constructed only in certain ways.  For this purpose, Dafny
+To write structured object-oriented programs, one often relies on 
+objects being constructed only in certain ways.  For this purpose, Dafny
 provides _constructor (method)s_, which are a restricted form of
 initialization methods.  A constructor is declared with the keyword
-`constructor` instead of `method`.  When a class contains a
-constructor, every call to `new` for that class must be accompanied
-with a call to one of the constructors.  Moreover, a constructor
+`constructor` instead of `method`.  
+When a class contains a
+constructor, every call to `new` for a class must be accompanied
+by a call to one of its constructors.  Moreover, a constructor
 cannot be called at other times, only during object creation.  Other
 than these restrictions, there is no semantic difference between using
-ordinary initialization methods and using constructors.
-
-The Dafny design allows the constructors to be named, which promotes
-using names like `InitFromList` above.  Still, many classes have just
+ordinary initialization methods and using constructors. Classes may
+declare no constructors or one or more constructors.
+
+#### Classes with no explicit constructors
+
+A class that declares no constructors has a default constructor created 
+for it. This constructor is called with the syntax
+```
+C C := new C;
+```
+This constructor simply initializes the fields of the class.
+The order of initialization is such that any field whose value depends on the
+value of another field is initialized after that other field. 
+It is an error if there are circular dependencies. 
```

Comment:
This is confusing without mention of `const` fields with a RHS. I suggest something along the lines of:

> The declaration of a `const` field may include a right-hand side (RHS) that specifies the constant's value. The RHS of a `const` field may depend on each constant fields, but circular dependencies are not allowed.

Note that the RHS of a `const` may depend on constant fields without a RHS. For example,

    const a := b;
    const b: int

defines `a` to have the same value as `b`. The value of `b` is set arbitrarily (if the class has no constructors) or by a constructor (if the class has one or more constructors).

Then, I suggest prefixing L1946 with "Except for `const` fields with a RHS".

Reply:
Reworded